x-pack/test/api_integration/apis/spaces/set_solution_space.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const spacesService = getService('spaces');
+
+  describe('PUT /internal/spaces/space/{id}/solution', () => {
+    before(async () => {
+      await spacesService.create({
+        id: 'foo-space',
+        name: 'Foo Space',
+        disabledFeatures: [],
+        color: '#AABBCC',
+      });
+    });
+
+    after(async () => {
+      await spacesService.delete('foo-space');
+    });
+
+    it('Use solution_type param to set solution for default space', async () => {
+      await supertest
+        .put('/internal/spaces/space/default/solution')
+        .set('kbn-xsrf', 'xxx')
+        .set('x-elastic-internal-origin', 'cloud')
+        .send({
+          solution_type: 'observability',
+        })
+        .expect(200)
+        .then((response) => {
+          const { solution, name, id } = response.body;
+          expect({ solution, name, id }).to.eql({
+            id: 'default',
+            name: 'Default',
+            solution: 'oblt',
+          });
+        });
+    });
+
+    it('Use solution_type param to set solution for Foo Space space', async () => {
+      await supertest
+        .put('/internal/spaces/space/foo-space/solution')
+        .set('kbn-xsrf', 'xxx')
+        .set('x-elastic-internal-origin', 'cloud')
+        .send({
+          solution_type: 'observability',
+        })
+        .expect(200)
+        .then((response) => {
+          const { solution, name, id } = response.body;
+          expect({ solution, name, id }).to.eql({
+            id: 'foo-space',
+            name: 'Foo Space',
+            solution: 'oblt',
+          });
+        });
+    });
+
+    it('Use solution param to set solution for default space', async () => {
+      await supertest
+        .put('/internal/spaces/space/default/solution')
+        .set('kbn-xsrf', 'xxx')
+        .set('x-elastic-internal-origin', 'cloud')
+        .send({
+          solution: 'oblt',
+        })
+        .expect(200)
+        .then((response) => {
+          const { solution, name, id } = response.body;
+          expect({ solution, name, id }).to.eql({
+            solution: 'oblt',
+            id: 'default',
+            name: 'Default',
+          });
+        });
```

Comment:
__nit:__ If I read the code correctly, we don't reset `solution` for the custom and default space, so technically, this test and the test below will pass even if this call doesn't do anything (since it was set to `oblt` by previous tests). Would it be possible to reset `solution` to `classic` for both the default and custom spaces in `afterEach`?